multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/CheckForOperationsInProgressStep.java

```diff
@@ -30,89 +28,60 @@
         }
 
         getStepLogger().info(Messages.WAITING_PREVIOUS_OPERATIONS_TO_FINISH);
 
-        Map<String, ServiceOperation.Type> servicesOperationTypes = getServicesOperationTypes(servicesInProgressState);
+        Map<String, ServiceOperation.Type> servicesOperationTypes = getServicesOperationTypes(existingServicesInProgress);
         getStepLogger().debug(Messages.SERVICES_IN_PROGRESS, JsonUtil.toJson(servicesOperationTypes, true));
         context.setVariable(Variables.TRIGGERED_SERVICE_OPERATIONS, servicesOperationTypes);
 
-        List<CloudServiceInstanceExtended> servicesWithData = getListOfServicesWithData(servicesInProgressState);
-        context.setVariable(Variables.SERVICES_DATA, servicesWithData);
+        context.setVariable(Variables.SERVICES_DATA, existingServicesInProgress);
 
         return StepPhase.POLL;
     }
 
-    protected List<CloudServiceInstanceExtended> getServicesToProcess(ProcessContext context) {
-        return Collections.singletonList(context.getVariable(Variables.SERVICE_TO_PROCESS));
-    }
-
-    private List<CloudServiceInstanceExtended> getExistingServices(CloudControllerClient cloudControllerClient,
-                                                                   List<CloudServiceInstanceExtended> servicesToProcess) {
-        return servicesToProcess.parallelStream()
-                                .map(service -> getExistingService(cloudControllerClient, service))
-                                .filter(Objects::nonNull)
-                                .collect(Collectors.toList());
+    protected List<CloudServiceInstanceExtended> getExistingServiceInProgress(ProcessContext context) {
+        CloudControllerClient client = context.getControllerClient();
+        CloudServiceInstanceExtended serviceToProcess = context.getVariable(Variables.SERVICE_TO_PROCESS);
+        CloudServiceInstanceExtended existingServiceInstance = getExistingService(client, serviceToProcess);
+        if (existingServiceInstance == null || !isServiceOperationInProgress(existingServiceInstance)) {
+            return Collections.emptyList();
+        }
+        return List.of(existingServiceInstance);
     }
 
-    private CloudServiceInstanceExtended getExistingService(CloudControllerClient cloudControllerClient,
-                                                            CloudServiceInstanceExtended service) {
+    protected CloudServiceInstanceExtended getExistingService(CloudControllerClient cloudControllerClient,
+                                                              CloudServiceInstanceExtended service) {
         CloudServiceInstance existingService = cloudControllerClient.getServiceInstance(service.getName(), false);
         if (existingService != null) {
             return ImmutableCloudServiceInstanceExtended.builder()
                                                         .from(service)
+                                                        .from(existingService)
```

Comment:
This is intended, as I want to merge the data from both the cached service object and the one coming from CC.
The cached one is `CloudServiceInstanceExtended` and might have additional information than the one returned from CC (which is only `CloudServiceInstance`.

Reply:
Won't the data from `service` be overwritten with the data from `existingService`?

Reply:
That is what is intended